prometheus_hardware_exporter/__main__.py

```diff
@@ -77,17 +104,40 @@ def parse_command_line() -> argparse.Namespace:
     return args
 
 
-def main(port: int, level: str, enable_collectors: List[str], daemon: bool = False) -> None:
+# pylint: disable=R0913
+def main(
+    port: int,
+    level: str,
+    enable_collectors: List[str],
+    redfish_host: str,
+    redfish_username: str,
+    redfish_password: str,
+    daemon: bool = False,
+) -> None:  # pragma: no cover
     """Start the prometheus-hardware-exporter."""
     root_logger = logging.getLogger()
     root_logger.setLevel(logging.getLevelName(level))
 
     exporter = Exporter(port)
     enable_collectors_set = set(enable_collectors)
-    for name, collector in COLLECTOR_REGISTRIES.items():
-        if name.lower() in enable_collectors_set:
-            logger.info("%s is enabled", name)
-            exporter.register(collector)
+    if "collector.hpe_ssa" in enable_collectors_set:
```

Comment:
These `if` are awful. I guess you misunderstand what I mean in 
https://github.com/canonical/prometheus-hardware-exporter/pull/16#discussion_r1246361637